libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -676,7 +683,24 @@ extern "C" {
     char res[20];
     br_sha1_out(&xc->sha1_cert, res);
     if (xc->match_fingerprint && memcmp(res, xc->match_fingerprint, sizeof(res))) {
+#ifdef DEBUG_ESP_SSL
       DEBUG_BSSL("insecure_end_chain: Received cert FP doesn't match\n");
+      char buff[3 * sizeof(res) + 5];
+      buff[0] = 0;
+      for (size_t i=0; i<sizeof(res); i++) {
+        char hex[6];
```

Comment:
Why 6? What is the max possible output of sprintf below? What about nullterm? Shouldn't snprintf and strncat be used instead?

I suspect the max output of sprintf is 4 bytes: 2 hex digits a space and a nullterm. Despite knowing the max size, it's considered best to move towards de "n" versions of the classic functions.

Reply:
Updated in latest push.